File: Assembler/LB 4/Task2.c
#include <stdio.h>

// f(x) = 5x^3 - x^2 + 6 Написать подпрограмму, вычисляющую значение функции 4f(x1) + f(x2)

int f(int x) {
    return 5 * x * x * x - x * x + 6;
}

int main(void) {
    int x1 = 1, x2 = 5;
    int f1 = 4 * f(x1);
    int f2 = f(x2);
    int result = f1 + f2;
    printf("4f(x1) + f(x2) = %d + %d = %d\n", f1, f2, result);
}

File: Programming Methods/lab8/Task2.py
from typing import List
import random

def generate_matrix(M: int, N: int, seed: int = None) -> List[List[int]]:
    if seed:
        random.seed(seed)
    return [[random.randint(-50, 51) for _ in range(N)] for _ in range(M)]

def find_matrix_sum(matrix: List[List[int]]):
    return sum(sum(line) for line in matrix)

def get_negative_elements(matrix: List[List[int]]) -> List[tuple]:
    result = []
    for i in range(len(matrix)):
        for j in range(len(matrix[i])):
            if matrix[i][j] < 0:
                result += [(matrix[i][j], i, j)]
    return sorted(result, key=lambda x: x[0])

def zero_out(matrix: List[List[int]], negatives: List[tuple]):
    i, j = negatives[0][1], negatives[0][2]
    negatives.pop(0)
    matrix[i][j] = 0

def main():
    matrix = generate_matrix(5, 5, 2011)
    print(*matrix, sep="\n")
    negatives = get_negative_elements(matrix)
    success = False
    while True:
        if find_matrix_sum(matrix) > 0:
            success = True
            break
        if len(negatives) == 0:
            break
        zero_out(matrix, negatives)
    print()
    print(*matrix, sep="\n") if success else print("Невозможно сделать сумму положительной.")
        
if __name__  == "__main__":
    main()